cmd/awscollector/Dockerfile

```diff
@@ -1,19 +1,59 @@
+# The binary source type (options: build, copy)
+ARG BIN_SRC_TYPE=build
+
 ################################
-#	Building Stage         #	
-#			       #	
+#	Certificate Stage      #
+#			       #
 ################################
-FROM alpine:latest as build 
+FROM alpine:latest AS certs
 
-# update cert
 RUN apk --update add ca-certificates
 
+################################
+#	Building Stage         #
+#			       #
+################################
+FROM golang:1.17 AS prep-build
+
+ENV GOPROXY=direct
+
+ARG TARGETARCH
+
+# download go modules ahead to speed up the building
+WORKDIR /workspace
+COPY go.mod .
+COPY go.sum .
+RUN go mod download
```

Comment:
Will this take a long time. @anuraaga suggestion was to move this another dockerfile.

Reply:
Have 1 for coping the binary and 1 where you can build the binary inside of the build container

Reply:
This is just a way to have both in a single Dockerfile. We typically won't use the build option, so the stage will be skipped, but it's there for those that want it.

Reply:
Some documentation/description of that in this file might be good.  It's a bit opaque.

Reply:
Added some documentation at the top of the Dockerfile where the ARG is declared.